SAMtemplates/main_template.yaml

```diff
@@ -0,0 +1,337 @@
+Globals:
+  Function:
+    Timeout: 50
+    MemorySize: 256
+    Architectures:
+      - x86_64
+    Runtime: nodejs18.x
+    Environment:
+      Variables:
+        TargetServiceSearchServer: !Ref TargetServiceSearchServer
+        NODE_OPTIONS: --enable-source-maps
+    Layers:
+      - !Sub "arn:aws:lambda:${AWS::Region}:580247275435:layer:LambdaInsightsExtension:38"
+
+Parameters:
+  TargetServiceSearchServer:
+    Type: String
+    Description: TargetServiceSearchServer
+    Default: none
+  TruststoreVersion:
+    Type: String
+    Description: TruststoreVersion
+    Default: none
+  EnableMutualTLS:
+    Type: String
+    Description: Whether to use mutual TLS
+    Default: false
+    AllowedValues: [true, false]
+  EnableSplunk:
+    Type: String
+    Description: Whether to use splunk
+    Default: false
+    AllowedValues: [true, false]
+  LogLevel:
+    Type: String
+    Description: The log level to set in the lambda
+    Default: "INFO"
+  LogRetentionDays:
+    Type: Number
+    Description: How long to keep logs for
+    Default: 30
+    AllowedValues:
+      [
+        1,
+        3,
+        5,
+        7,
+        14,
+        30,
+        60,
+        90,
+        120,
+        150,
+        180,
+        365,
+        400,
+        545,
+        731,
+        1096,
+        1827,
+        2192,
+        2557,
+        2922,
+        3288,
+        3653,
+      ]
+
+Conditions:
+  ShouldUseMutualTLS: !Equals [true, !Ref EnableMutualTLS]
+  ShouldUseSplunk: !Equals [true, !Ref EnableSplunk]
+
+Resources:
+  # updatePrescriptionStatus lambda
+  UpdatePrescriptionStatusResources:
+    Type: AWS::Serverless::Application
+    Properties:
+      Location: lambda_resources.yaml
+      Parameters:
+        CloudWatchKMSKey: !ImportValue account-resources:CloudwatchLogsKmsKeyArn
+        LambdaName: !Sub "${AWS::StackName}-UpdatePrescriptionStatus"
+        LogRetentionDays: !Ref LogRetentionDays
+  UpdatePrescriptionStatus:
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: !Sub "${AWS::StackName}-UpdatePrescriptionStatus"
+      CodeUri: ../packages
+      Handler: 'updatePrescriptionStatus.handler'
+      Role: !GetAtt UpdatePrescriptionStatusResources.Outputs.LambdaRoleArn
+      Environment:
+        Variables:
+          TABLE_NAME: !Sub '${AWS::StackName}-PrescriptionStatusUpdate'
+          LOG_LEVEL: !Ref LogLevel
+      Events:
+        updatePrescriptionStatus:
+          Type: HttpApi
+          Properties:
+            ApiId: !Ref HttpApiGateway
+            Path: /
+            Method: post
+    Metadata: # Manage esbuild properties
+      BuildMethod: esbuild
+      BuildProperties:
+        Minify: true
+        Target: "es2020"
+        Sourcemap: true
+        tsconfig: updatePrescriptionStatus/tsconfig.json
+        EntryPoints:
+          - updatePrescriptionStatus/src/updatePrescriptionStatus.ts
+
+  # Prescription Status Update table
+  PrescriptionStatusUpdate:
```

Comment:
Resource name could probably include Table, so its more obvious what it is as its current name is also the name of the overall service.

Reply:
The resource has been renamed to PrescriptionStatusUpdatesTable.